client/my-sites/media-library/media-date-range.jsx

```diff
@@ -0,0 +1,344 @@
+		};
+
+		const rawDay = this.momentDateToNative( date );
+
+		// Calculate the new Date range
+		const newRange = DateUtils.addDayToRange( rawDay, range );
+
+		// Edge case: Range can sometimes be from: null, to: null
+		if ( ! newRange.from || ! newRange.to ) return;
+
+		this.setState(
+			previousState => {
+				let newState = {
+					startDate: this.nativeDateToMoment( newRange.from ),
+					endDate: this.nativeDateToMoment( newRange.to ),
+					inputFromDate: this.nativeDateToMoment( newRange.from ).format( 'L' ),
+					inputToDate: this.nativeDateToMoment( newRange.to ).format( 'L' ),
+				};
+
+				// For first date selection only: "cache" previous dates
+				// just in case user doesn't "Apply" and we need to revert
+				// to the original dates
+				if ( ! this.state.oldDatesSaved ) {
+					newState = Object.assign( {}, newState, {
+						oldStartDate: previousState.startDate,
+						oldEndDate: previousState.endDate,
+						oldDatesSaved: true, // marks that we have saved old dates
+					} );
+				}
+
+				return newState;
+			},
+			() => this.props.onDateSelect( this.state.startDate, this.state.endDate )
+		);
+	}
+
+	revertDates() {
+		this.setState( previousState => {
+			const newState = { oldDatesSaved: false };
+
+			if ( previousState.oldStartDate && previousState.oldEndDate ) {
+				newState.startDate = previousState.oldStartDate;
+				newState.endDate = previousState.oldEndDate;
+			}
+
+			return newState;
+		} );
+	}
+
+	commitDates() {
+		this.setState(
+			previousState => ( {
+				oldStartDate: previousState.startDate, // update cached old dates
+				oldEndDate: previousState.endDate, // update cached old dates
+				oldDatesSaved: false,
+			} ),
+			() => {
+				this.props.onDateCommit( this.state.startDate, this.state.endDate );
+				this.togglePopover();
+			}
+		);
+	}
+
+	momentDateToNative( momentDate ) {
+		return momentDate.toDate();
+	}
+
+	nativeDateToMoment( nativeDate ) {
+		return this.props.moment( nativeDate );
+	}
+
+	dateToHumanReadable( date ) {
+		return this.props.moment( date ).format( 'L' );
+	}
+
+	renderDatePicker() {
+		const now = new Date();
+
+		return (
+			<DatePicker
+				className="media-library__date-range-popover-date-picker"
+				showOutsideDays={ false }
+				toMonth={ now }
+				onSelectDay={ this.onSelectDate }
+				selectedDays={ {
+					from: this.momentDateToNative( this.state.startDate ),
+					to: this.momentDateToNative( this.state.endDate ),
+				} }
+				numberOfMonths={ window.matchMedia( '(min-width: 480px)' ).matches ? 2 : 1 }
+				calendarViewDate={ this.momentDateToNative( this.state.startDate ) }
+				disabledDays={ [
+					{
+						after: now, // you can't look at photos from the future!
+					},
+				] }
+			/>
+		);
+	}
+
+	renderPopoverHeader() {
+		return (
+			<div className="media-library__date-range-popover-header">
+				<Button
+					className="media-library__date-range-popover-cancel-btn"
+					onClick={ this.togglePopover }
```

Comment:
It looks like this is another case where we should be naming the `togglePopover` method `closePopover` or similar.
Also, if I follow correctly, it seems we might be doing extra work here as when we click this 'Cancel' button we would be calling 
```js
this.props.onDateCommit( this.state.startDate, this.state.endDate );
```
via `togglePopover`

Reply:
> It looks like this is another case where we should be naming the togglePopover method closePopover or similar.

👍 

> Also, if I follow correctly, it seems we might be doing extra work here

I need to look into this again. The aim is to _ignore_ any dates the user has selected and revert both the _committed_ and the _selected_ dates back to the original dates the user selected _before_ they opened the popover.

Reply:
Created dedicated `open` and `close` methods in https://github.com/Automattic/wp-calypso/pull/29938